File: overthewire/semtex/semtex0.py
#!/usr/bin/python

import socket
import os, sys, stat

con = socket.socket(socket.AF_INET ,socket.SOCK_STREAM)
address = "www.semtex.labs.overthewire.org"
PORT = 24001
print "Trying to connect to " + str(address) + " on port "+ str(PORT) +"..."
con.connect((address, PORT))
print "Connected!"

data = ""
counter = 1

while True:
	good = con.recv(1)
	data += good
	dump = con.recv(1)
	if not good or not dump:
		break
	counter+=1

print "Recived " + str(counter) + " Bytes!"

con.close()

print "Closed Connection!"
print "Starting to write exe..."

out = open("semtex0py" ,"w")
out.write(data)
out.close()

print "Finished writing!"
print "Executing..."

os.chmod("semtex0py",0700)

print "Output:"
os.system("./semtex0py")
os.remove("semtex0py")

print ""
print ""
print "Done!"

File: overthewire/semtex/semtex0.php
#!/usr/bin/php

<?php
	echo "Starting...\n";
	
    error_reporting(E_ALL);

    $address = gethostbyname('www.semtex.labs.overthewire.org');
    define("PORT", 24001);

	$socket = socket_create (AF_INET ,SOCK_STREAM ,SOL_TCP);

	if($socket === false)
	{
		echo "socket_create() failed: reason: " . socket_strerror(socket_last_error($socket)) . "\n";
		exit;
	}

	echo "Trying to connect to ". $address ." on port ". PORT ."\n";
	if(!socket_connect($socket , $address ,PORT))
	{
		echo "socket_connect() failed: reason: " . socket_strerror(socket_last_error($socket)) . "\n";
		exit;
	}

	$counter = 1;
	$temp = "";
	$buf = "";

	do
	{
		$buf .= $temp;
		$counter++;
	}
	while(($res = socket_recv($socket ,$temp ,1 ,0)) !== false && $res !== 0);

	echo "Recived ".$counter." Bytes\n\n";

	$exe = fopen("semtex0",'w');
	if($exe === false)
	{
		echo "Couldn't open executable\n";
		exit;
	}

	$message = str_split($buf);
	for($i = 0, $j = 1; $i < count($message);$i += 2, $j += 2)
	{
		if(fwrite($exe, $message[$i]) === false)
		{
			echo "Couldn't write to file, byte #". ($i/2). "\n";
			exit;
		}
}

	fclose($exe);
	chmod("semtex0",0700);
	system("./semtex0");

	echo "\n\nDone\n";
	exit;
?>